builder_test.go

```diff
@@ -209,6 +210,38 @@ RUN echo $FOO $BAR`))
 	}
 }
 
+// Test if `FROM some-${SOME-BUILT-IN-ARG}` args gets resolved correctly.
+func TestArgResolutionOfDefaultVariables(t *testing.T) {
+	// Get architecture from host
+	var localspec = platforms.DefaultSpec()
+	for _, tc := range []struct {
+		name         string
+		args         map[string]string
+		expectedFrom string
+	}{
+		{name: "override", args: map[string]string{"FOO": "bar"}, expectedFrom: "platform-"+localspec.Architecture},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			n, err := ParseDockerfile(strings.NewReader(`ARG FOO=latest
+ARG BAR=baz
+FROM platform-${TARGETARCH}
+ARG BAZ=banana
+RUN echo $FOO $BAR`))
```

Comment:
This can probably be simplified from the version in the `TestHeadingArg` test, since the values for FOO, BAR, and BAZ aren't used on the FROM line.

Reply:
Yes we don't need extra lines only lines in the format of `FROM ${ARG}` is enough for this test.